mitodl/open-discussions
# Add url to Course model and helper method to determine its value

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #1834 

#### What's this PR do?
- Adds `url` to the `Course` model
- Adds `task_helpers.get_course_url` to determine what that value should be, which is unfortunately not very straightforward (see #1834 for details).

#### How should this be manually tested?
- Run the new migration.  If you already have courses imported, they should have url values afterward.
- If you do not already have courses imported, run `recreate_index`, set `OCW_UPLOAD_IMAGE_ONLY=True` in your `.env` file (and set other `EDX_` and `OCW_` variables) and then `backpopulate_edx_data` and `backpopulate_ocw_data` - you can interrupt the ocw import after a few minutes, it will take hours to import all.  Verify that the courses have url's.
